src/settings/target/mod.rs

```diff
@@ -208,7 +193,7 @@ impl Manifest {
 
         self.check_private(environment);
 
-        let (route, workers_dev) = self.negotiate_zoneless(environment, release)?;
+        let (route, workers_dev) = self.negotiate_zoneless(environment)?;
```

Comment:
i wonder if we could benefit from a type here that's just `DeployConfig`; an enum that can be either Zoneless, or a struct, Zoned, with a zone_id and route, and include it as `target.deploy_config` rather than three attributes of `target` that we incessantly check.

Reply:
I like this - but I think I like Gabbi's approach of making Route an Option instead, gonna do that

Reply:
we could likely do both, but mine is a bit more... involved. gabbi's is likely the better answer